src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,46 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] diagonal = new int[matrix.length];
+        for (int i = 0, j = 0; i < matrix.length; i++, j++) {
+            if (i == j) {
```

Comment:
this check here is redundant, cause it's square matrix according to description and you increment i and j at the same time

Reply:
removed